include/seqan3/range/view/drop.hpp

```diff
@@ -0,0 +1,204 @@
+{
+private:
+    //!\brief Type of the CRTP-base.
+    using base_t = pipable_adaptor_base<drop_fn>;
+
+public:
+    //!\brief Inherit the base class's Constructors.
+    using base_t::base_t;
+
+private:
+    //!\brief Befriend the base class so it can call impl().
+    friend base_t;
+
+    /*!\brief       Call the view's constructor with the underlying view as argument.
+     * \returns     An instance of seqan3::detail::view_drop.
+     */
+    template <std::ranges::ViewableRange urng_t>
+    static auto impl(urng_t && urange, size_t drop_size)
+    {
+        auto b = std::ranges::begin(urange);
+
+        if constexpr (std::ranges::SizedRange<urng_t>)
+        {
+            drop_size = std::min(drop_size, std::ranges::size(urange));
+            std::ranges::advance(b, drop_size);
+
+            return std::ranges::subrange<std::ranges::iterator_t<urng_t>,
+                                         std::ranges::sentinel_t<urng_t>,
+                                         std::ranges::subrange_kind::sized>
+            {
+                std::move(b),
+                std::ranges::end(urange),
+                std::ranges::size(urange) - drop_size
+            };
+        }
+        else
+        {
+            for (size_t i = 0; (i < drop_size) && (b != std::ranges::end(urange)); ++i, ++b);
+
+            return std::ranges::subrange<std::ranges::iterator_t<urng_t>,
+                                         std::ranges::sentinel_t<urng_t>>
+            {
+                std::move(b),
+                std::ranges::end(urange)
+
+            };
+        }
+    }
+
+    /*!\brief       Overload for contiguous, sized ranges.
+     * \returns     A std::span over the input.
+     */
+    template <std::ranges::ViewableRange urng_t>
+        requires std::ranges::ContiguousRange<urng_t> && std::ranges::SizedRange<urng_t>
+    static auto impl(urng_t && urange, size_t drop_size)
+    {
+        drop_size = std::min(drop_size, std::ranges::size(urange));
+        return std::span{std::ranges::data(urange) + drop_size, std::ranges::size(urange) - drop_size};
+    }
+
+    /*!\brief       Overload for std::basic_string_view.
+     * \returns     A std::span over the input.
+     */
+    template <std::ranges::ViewableRange urng_t>
+        requires std::ranges::ContiguousRange<urng_t> && std::ranges::SizedRange<urng_t> &&
+                 is_type_specialisation_of_v<std::remove_reference_t<urng_t>, std::basic_string_view>
+    static auto impl(urng_t && urange, size_t drop_size)
+    {
+        drop_size = std::min(drop_size, std::ranges::size(urange));
+        return urange.substr(drop_size);
+    }
+
+    /*!\brief       Overload for std::basic_string.
+     * \returns     A std::span over the input.
+     */
+    template <typename char_t, typename traits_t, typename allocator_t>
+    static std::basic_string_view<char_t, traits_t> impl(std::basic_string<char_t, traits_t, allocator_t> & urange,
```

Comment:
hm it is sufficient here for the overload resolution to give the explicit type basic_string but above you require `is_type_specialisation_of_v<std::basic_string_view>`...?

Reply:
above I want all four overloads, that's why I use the forwarding reference. Below I want to explicitly forbid taking rvalues and I need different implementations for const and non-const.